src/main/kotlin/uk/gov/justice/digital/hmpps/prison/visitbooker/registry/service/AuthService.kt

```diff
@@ -32,10 +32,10 @@ class AuthService(
   private fun match(authDetail: AuthDetail): Booker {
     var booker: Booker
     if (authDetail.count == 0) {
-      booker = bookerRepository.findByEmail(authDetail.email) ?: throw BookerNotFoundException("Booker for Email : ${authDetail.email} not found")
+      booker = bookerRepository.findByEmailIgnoreCase(authDetail.email) ?: throw BookerNotFoundException("Booker for Email : ${authDetail.email} not found")
       booker.oneLoginSub = authDetail.oneLoginSub
       // Create reference and then save
-      if (booker.reference.isBlank()) {
+      if (booker.reference.isNullOrBlank()) {
```

Comment:
updated to isNullOrBlank as the DB supports reference being null and it was throwing a NullPointerException while testing locally.

Reply:
Sorry this was my fault. Intellij complained as the DTO contained a NotNull validation annotation, and I thought it wasn't needed. My bad!

Reply:
no worries. Yes the IDE is showing it as not needed.